@Handler on Bean does not resolve method

Hi Team.

Busy working on another example and came across the following error when using the ```@Handler``` on a bean.

Route below:
```
    @Override
    public void configure() throws Exception {
        camelContext.setMessageHistory(true);
        fromF("timer:foo?period=%s", period)
                .to("bean:greeterBean")
                .to("log:example");

    }
```

The bean definition:

```
@ApplicationScoped
@Named("greeterBean")
@RegisterForReflection
public class Greeter {

    public String greet(Exchange exchange) {
        return "greet hello world";
    }

    @Handler
    public String say(Exchange exchange) {
        return "say hello world";
    }
}
```

The following exception is thrown.

```
 org.apache.camel.component.bean.AmbiguousMethodCallException: Ambiguous method invocations possible: [public java.lang.String org.acme.sedaInterceptLog.Greeter_ClientProxy.greet(org.apache.camel.Exchange), public java.lang.String org.acme.sedaInterceptLog.Greeter_ClientProxy.say(org.apache.camel.Exchange)] on the exchange: Exchange[]
        at org.apache.camel.component.bean.BeanInfo.chooseMethod(BeanInfo.java:614)
```

Seems like ```@Handler``` is ignored completely.

If I change the bean definition to:

```
  
@ApplicationScoped
@Named("greeterBean")
@RegisterForReflection
public class Greeter {

    @Handler
    public String say() {
        return "say hello world";
    }
}
```

I get the following exception:
```
Caused by: [java.lang.IllegalStateException - No method invocation could be created, no matching method could be found on: org.acme.sedaInterceptLog.Greeter@946f731]
```

Is the ```@Handler``` annotation going to be used going forward? I can get the example to work just wondering about the annotations going forward.

Hey @lburgazzoli 

I have tried using the static binding method as you have detailed in the open pull [1461](https://github.com/apache/camel-quarkus/pull/1461  ) however this is not working on my side.

I am getting the following exception:

```Caused by: org.apache.camel.ResolveEndpointFailedException: Failed to resolve endpoint: bean://withHandler due to: No bean could be found in the registry for: withHandler```


I try and reproduce with your example later today but in my example it is not working at the moment.

--- a/integration-tests/bean/src/main/java/org/apache/camel/quarkus/component/bean/CamelResource.java
+++ b/integration-tests/bean/src/main/java/org/apache/camel/quarkus/component/bean/CamelResource.java
@@ -71,6 +71,14 @@ public class CamelResource {
         return template.requestBody("direct:handler", statement, String.class);
     }
 
+    @Path("/handlerOnProxy")
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.TEXT_PLAIN)
+    public String handlerOnProxy(String statement) {
+        return template.requestBody("direct:handlerOnProxy", statement, String.class);
+    }
+
     @Path("/increment")
     @GET
     @Produces(MediaType.TEXT_PLAIN)

--- a/integration-tests/bean/src/main/java/org/apache/camel/quarkus/component/bean/CamelRoute.java
+++ b/integration-tests/bean/src/main/java/org/apache/camel/quarkus/component/bean/CamelRoute.java
@@ -76,6 +76,10 @@ public class CamelRoute extends RouteBuilder {
                 .to("bean:withHandler")
                 .to("log:named");
 
+        from("direct:handlerOnProxy")
+                .to("bean:withHandlerOnProxy")
+                .to("log:named");
+
     }
 
     @SuppressWarnings("unchecked")

--- a/integration-tests/bean/src/main/java/org/apache/camel/quarkus/component/bean/WithHandlerBean.java
+++ b/integration-tests/bean/src/main/java/org/apache/camel/quarkus/component/bean/WithHandlerBean.java
@@ -16,10 +16,15 @@
  */
 package org.apache.camel.quarkus.component.bean;
 
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Named;
+
 import io.quarkus.runtime.annotations.RegisterForReflection;
 import org.apache.camel.Exchange;
 import org.apache.camel.Handler;
 
+@ApplicationScoped
+@Named("withHandlerOnProxy")
 @RegisterForReflection
 public class WithHandlerBean {
     /**

--- a/integration-tests/bean/src/test/java/org/apache/camel/quarkus/component/bean/BeanTest.java
+++ b/integration-tests/bean/src/test/java/org/apache/camel/quarkus/component/bean/BeanTest.java
@@ -69,6 +69,16 @@ public class BeanTest {
     }
 
     @Test
+    public void handlerWithProxy() {
+        RestAssured.given()
+                .contentType(ContentType.TEXT)
+                .body("Kermit")
+                .post("/bean/handlerOnProxy")
+                .then()
+                .body(equalTo("Hello Kermit from the WithHandlerBean"));
+    }
+
+    @Test
     public void inject() {
 
         /* Ensure that @Inject works */

